aws-lc-fips-sys/builder/cmake_builder.rs

```diff
@@ -93,6 +93,16 @@ impl CmakeBuilder {
     const GOCACHE_DIR_NAME: &'static str = "go-cache";
     #[allow(clippy::too_many_lines)]
     fn prepare_cmake_build(&self) -> cmake::Config {
+        // Set the compiler to clang if CC/CXX are not set
+        // TODO: this is a workaround until aws-lc fixes build on GGC-14
+        // See https://github.com/aws/aws-lc/issues/201
+        if env::var("CC").is_err() {
+            env::set_var("CC", "clang");
+        }
```

Comment:
If clang is not available on the system, then this would likely cause a build failure. 

I think switching the compiler on the user is something that should only be done when (1) we know clang is available in the environment, and (2) the switch is necessary to avoid an inevitable build failure. Even in that case, quietly switching which compiler is used bothers me a little. Would it be better to instead instruct the user to set the environment variable themselves?

Reply:
The problem with setting the env vars globally is that it imposes change to the build of other crates that also use the 'cc' crate.

I.e what if a crate "foo" has a bug when building with clang and now I'm deadlocked and cannot build both aws-lc-sys-rs and foo with the same global compiler env var?


I agree that failing the build if clang does not exist may not be nice for the user.

I can't find a way to tell the cc crate to "prefer clang if installed and env vars are not set"

Reply:
I added a workaround to address this on [another PR](https://github.com/aws/aws-lc-rs/pull/678): https://github.com/aws/aws-lc-rs/pull/678/commits/81c9f00d5afa4c30e634c500eb6556254e51311c

Reply:
Ah, wonderful, we can close this one and I'll patiently await for it to be released :)